debian/rules

```diff
@@ -69,6 +73,11 @@ override_dh_auto_install:
 	# We want to guarantee that we are not shipping any conftest files
 	find $(CURDIR)/debian/ubuntu-advantage-tools -type f -name conftest.py -delete
 
+ifneq (${VERSION_ID},"16.04")
+	# Only install cloud-id-shim on Xenial
+	rm $(CURDIR)/debian/ubuntu-advantage-tools/lib/systemd/system/ubuntu-advantage-cloud-id-shim.service
```

Comment:
Maybe I am missing something here, but the idea would be to remove the `shim` if we upgrade from Xenial right ? If so, shouldn't the command be a `rm -f` in case we are already on a later version of Ubuntu ?

Also, since this is a systemd unit, don't we need to remove it similarly to what we do with the old timer ?

Reply:
My understanding is that this runs during package creation (e.g. during `sbuild`) and this command removes the shim .service file from the package when building for non-xenial.

So then the Xenial package will have the file, so it will install/uninstall it.

Non-xenial packages will not have the file, so they won't even install it.

Note this is similar to how we're preventing `conftest` from getting into our deb packages

Reply:
Okay, that makes sense and answer my first question perfectly. But regarding a Xenial machine were the cloud-id-shim service is up and running. If we upgrade it to Bionic, wouldn't just removing the service through `rm` cause a weird state when running `systemctl status` ?

Reply:
I'm also going to move this branch to the canonical repo to get that sweet CI

Reply:
I tested this and couldn't find any errors.